app/models/Sort.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Indexes.{ascending, descending}
+
+sealed trait Sort {
+  val field: String
+  val orderBy: String
+  val convertParams: String
+  val toBson: Bson
+
+  override def toString: String = s"$field.$orderBy"
+}
+
+object Sort {
+
```

Comment:
Could maybe introduce an `Order` trait, then have `extends Sort with Ascending` for example.
```
import models.Sort.Order
import models.Sort.Order._

sealed trait Sort {
  self: Order =>

  val field: String
  val convertParams: String = this.toString

  override def toString: String = s"$field.$orderBy"
}

object Sort {

  sealed trait Order {
    val orderBy: String
  }

  object Order {
    sealed trait Ascending extends Order {
      override val orderBy: String = "asc"
    }
    sealed trait Descending extends Order {
      override val orderBy: String = "dsc"
    }
  }

  case object SortByLRNAsc extends Sort with Ascending {
    val field: String   = "lrn"
  }

  case object SortByLRNDesc extends Sort with Descending {
    val field: String   = "lrn"
  }

  case object SortByCreatedAtAsc extends Sort with Ascending {
    val field: String   = "createdAt"
  }

  case object SortByCreatedAtDesc extends Sort with Descending {
    val field: String   = "createdAt"
  }

  def apply(sortParams: Option[String]): Option[Sort] = sortParams map {
    case SortByLRNAsc.convertParams       => SortByLRNAsc
    case SortByLRNDesc.convertParams      => SortByLRNDesc
    case SortByCreatedAtAsc.convertParams => SortByCreatedAtAsc
    case _                                => SortByCreatedAtDesc
  }

}
```